### PR TITLE
[NP-9763] Fix capability lastModifiedBy/Agent table cell formatter

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-9763

## Changes
- Add reference property cell formatter
- Use ReferencePropertyCellFormatter for capability lastModifedBy/Agent

### CODE REVIEW

File: src/foam/u2/view/TableCellFormatter.js [hunk 1]
```diff
@@ -288,6 +288,30 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/TableCellFormatter.js:291]: Can't we just add this to the other reference formatter? Where it uses the property name if specified otherwise just uses the summary?

File: src/foam/u2/view/TableCellFormatter.js [hunk 2]
```diff
@@ -288,6 +288,30 @@ foam.CLASS({
...
+  name: 'ReferencePropertyCellFormatter',
+  implements: ['foam.u2.view.Formatter'],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'propName',
+      value: 'id',
+      documentation: 'Property on the reference object for display in tableview cell'
+    }
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/TableCellFormatter.js]:
```suggestion
    },
   {
      class: 'Boolean',
      name: 'projectionSafe'
    }
```

File: src/foam/nanos/crunch/Refinements.js [hunk 3]
```diff
@@ -31,18 +31,9 @@ foam.CLASS({
...
-          .select(obj.PROJECTION(foam.nanos.auth.User.LEGAL_NAME))
-          .then(function(result) {
-            if ( ! result || result.array.size < 1 || ! result.array[0]) {
-              this.add(value);
-              return;
-            }
-            this.add(result.array[0]);
-          }.bind(this));
+      tableCellFormatter: {
+        class: 'foam.u2.view.ReferencePropertyCellFormatter',
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/Refinements.js]: Why is this needed? Why not just use the existing ReferenceToSummaryCellFormatter, which will call to toSummary() method on User, and the first line of that is:
        if ( this.legalName ) return this.legalName;

Comment (lchanmann) [reply src/foam/nanos/crunch/Refinements.js]: ReferenceToSummaryCellFormatter doesn't show the value if the object is not found. 

Now updating ReferenceToSummaryCellFormatter https://github.com/kgrgreer/foam3/pull/2848/commits/0143a0669755dcb81dbf9fa64618fdea4767ea24 @kgrgreer @sarthak-marwaha

Comment (kgrgreer) [reply src/foam/nanos/crunch/Refinements.js]: Better to fix ReferenceToSummaryCellFormatter then.
